docs/src/tutorials/getting_started/debugging.jl

```diff
@@ -0,0 +1,270 @@
+# copies or substantial portions of the Software.                                #src
+#                                                                                #src
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR     #src
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,       #src
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE    #src
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER         #src
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,  #src
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE  #src
+# SOFTWARE.                                                                      #src
+
+# # Debugging
+
+# Dealing with bugs is an unavoidable part of coding optimization models in
+# JuMP. This includes bugs related to general Julia code such as syntax errors,
+# method errors, typos, and off-by-one indexing errors, but it also includes
+# optimization-specific bugs related to the formulation and solution of your
+# model.
+
+# This tutorial explains some common sources of bugs and modeling issues that
+# you might encounter when writing models in JuMP, and it suggests a variety of
+# strategies to deal with them.
+
+# !!! tip
+#     This tutorial is more advanced than the other "Getting started" tutorials.
+#     It's in the "Getting started" section to give you an early preview of how
+#     to test and debug JuMP models. However, if you are new to JuMP, you may
+#     want to briefly skim the tutorial, and come back to it once you have
+#     written a few JuMP models.
+
+using JuMP
+import HiGHS
+
+# ## Getting help
+
+# Debugging can be a frustrating part of modeling, particularly if you're new to
+# optimization and programming. If you're stuck, join the
+# [community forum](https://discourse.julialang.org/c/domain/opt/13) to search
+# for answers to commonly asked questions.
+
+# Before asking a new question, make sure to read the post
+# [make it easier to help you](https://discourse.julialang.org/t/please-read-make-it-easier-to-help-you/14757),
+# which contains a number of tips on how to ask a good question.
+
+# Above all else, take time to simplify your code as much as possible. The
+# fewer lines of code you can post that reproduces the same issue, the faster
+# someone can answer your question.
+
+# ## Debugging Julia code
+
+# Read the [Debugging chapter](https://benlauwens.github.io/ThinkJulia.jl/latest/book.html#chap21)
+# in the book [ThinkJulia.jl](https://benlauwens.github.io/ThinkJulia.jl/latest/book.html).
+# It has a number of great tips and tricks for debugging Julia code.
+
+# ## Debugging numerical issues
+
+# "Numerical issues" refers to a broad class of problems in which the solver
+# may return incorrect results, stall, fail to converge, or produce inconsistent
+# results.
```

Comment:
Perhaps give pointers to help diagnose the case where you have a real numerical error? E.g., relevant termination statuses. "Incorrect results" may be numerical issues or may be semantic (misspecification of the model). Alternatively, reframe the section around the termination statuses to make it more consistent with the discussions of infeasibility and unboundedness. Then, add a separate section on how to debug "incorrect results".

Reply:
How's this now?

Reply:
The relationship between the numerical issue section and the incorrect results section isn't clear. The sentence above still makes it seem like incorrect results is a subset of numerical issues. I'd refocus this section around "failure statuses" or "failure to solve" to reduce the overlap between sections. We don't even need to introduce the concept of "numerical issues", as it's not well defined.

Reply:
New structure looks good